core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java

```diff
@@ -76,6 +75,8 @@
 
     private final QueryManagerStats stats = new QueryManagerStats();
 
+    private DispatchQueryFactory dispatchQueryFactory;
```

Comment:
We should not be making this field mutable for a test.  I'll explain how in the test code below.